sdv/data_processing/data_processor.py

```diff
@@ -428,6 +427,14 @@ def update_transformers(self, column_name_to_transformer):
                 'updated.'
             )
 
+        for column, transformer in column_name_to_transformer.items():
+            if column in self._keys and not type(transformer) in (AnonymizedFaker, RegexGenerator):
+                raise InvalidPreprocessingError(
```

Comment:
I think according to our errors doc, this should be a `SynthesizerInputError` since it has more to do with a bad input than the process failing itself. In fact, I think that is the case for all instances of `InvalidPreprocessingError` in our code. We can just remove it. It's also not on our sheet at all. We were going to have a `SynthesizerPreprocessingError` as a catch all but I don't think we need it